src/yaml/yamlUtils.ts

```diff
@@ -0,0 +1,257 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: MIT
+ */
+
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import { Position } from 'vscode-languageserver-types'
+
+/* Returns an object with the following properties:
+    modifiedDocText: The text document derived from the document parameter which will be used to generate code completions
+    tempPositionForCompletions: The Position within modifiedDocText which will be used to generate code completions
+    startPositionForInsertion: The starting Position for the insert range applied to the generated code completions
+    endPositionForInsertion: The ending Position for the insert range applied to the generated code completions
+    shouldPrependSpace: A boolean indicating whether the generated code completions should have a space prepended to them
+*/
+export function processYamlDocForCompletion(document: TextDocument, position: Position): {
+    modifiedDocText: string,
+    tempPositionForCompletions: Position,
+    startPositionForInsertion: Position,
+    endPositionForInsertion: Position,
+    shouldPrependSpace: boolean
+} {
+    const lineOffsets = getLineOffsets(document.getText())
+    const { currentLine, currentLineEnd } = getCurrentLine(document, position, lineOffsets)
+
+    if (currentLine.indexOf(':') === -1) {
+        return processLineWithoutColon(document, position, currentLine, currentLineEnd)
+    } else {
+        return processLineWithColon(document, position, currentLine, currentLineEnd)
+    }
+}
+
+function processLineWithoutColon(document: TextDocument, cursorPosition: Position, currentLine: string, currentLineEnd: number): {
+    modifiedDocText: string,
+    tempPositionForCompletions: Position,
+    startPositionForInsertion: Position,
+    endPositionForInsertion: Position,
+    shouldPrependSpace: boolean
+} {
+    let modifiedDocText: string
+    let shouldPrependSpace: boolean = false
+
+    const tempPositionForCompletions: Position = {...cursorPosition}
+    const startPositionForInsertion: Position = {...cursorPosition}
+    const endPositionForInsertion: Position = {...cursorPosition}
+
+    // Since there's no colon to separate the key and value, replace all text after the cursor.
+    endPositionForInsertion.character = currentLine.length
+
+    const lineOffsets = getLineOffsets(document.getText())
+
+    const trimmedLine = currentLine.trim()
+
+    if (trimmedLine.length === 0) {
+        modifiedDocText =
+            // tslint:disable-next-line: prefer-template
```

Comment:
I actually think it would be more readable with template string. You can just create a few variables above.

Reply:
Updated to specify a preText, postText, and insertText which should clarify the text modifications we are doing.